ably/proto/message.go

```diff
@@ -22,12 +27,223 @@ type Message struct {
 	ClientID     string                 `json:"clientId,omitempty" codec:"clientId,omitempty"`
 	ConnectionID string                 `json:"connectionId,omitempty" codec:"connectionID,omitempty"`
 	Name         string                 `json:"name,omitempty" codec:"name,omitempty"`
-	Data         string                 `json:"data,omitempty" codec:"data,omitempty"`
+	Data         *DataValue             `json:"data,omitempty" codec:"data,omitempty"`
 	Encoding     string                 `json:"encoding,omitempty" codec:"encoding,omitempty"`
 	Timestamp    int64                  `json:"timestamp" codec:"timestamp"`
 	Extras       map[string]interface{} `json:"extras" codec:"extras"`
 }
 
+type DataValue struct {
```

Comment:
If this type is really necessary (see above), document the type and why/where it's used